app/code/Magento/SalesRule/Model/Converter/ToDataModel.php

```diff
@@ -5,77 +5,44 @@
+use Magento\SalesRule\Api\Data\RuleLabelInterfaceFactory;
 use Magento\SalesRule\Model\Data\Condition;
 use Magento\SalesRule\Api\Data\RuleInterface;
+use Magento\SalesRule\Model\Data\Condition as ConditionDataModel;
 use Magento\SalesRule\Model\Data\Rule as RuleDataModel;
 use Magento\SalesRule\Model\Rule;
 use Magento\Framework\Serialize\Serializer\Json;
+use Magento\SalesRule\Model\RuleFactory;
 
 class ToDataModel
 {
     /**
-     * @var \Magento\SalesRule\Model\RuleFactory
-     */
-    protected $ruleFactory;
-
-    /**
-     * @var \Magento\SalesRule\Api\Data\RuleInterfaceFactory
-     */
-    protected $ruleDataFactory;
-
-    /**
-     * @var \Magento\SalesRule\Api\Data\ConditionInterfaceFactory
-     */
-    protected $conditionDataFactory;
-
-    /**
-     * @var \Magento\Framework\Reflection\DataObjectProcessor
-     */
-    protected $dataObjectProcessor;
-
-    /**
-     * @var \Magento\SalesRule\Api\Data\RuleLabelInterfaceFactory
-     */
-    protected $ruleLabelFactory;
-
-    /**
-     * @var Json $serializer
-     */
-    private $serializer;
-
-    /**
-     * @var RuleExtensionFactory
-     */
-    private $extensionFactory;
-
-    /**
-     * @param \Magento\SalesRule\Model\RuleFactory $ruleFactory
-     * @param \Magento\SalesRule\Api\Data\RuleInterfaceFactory $ruleDataFactory
-     * @param \Magento\SalesRule\Api\Data\ConditionInterfaceFactory $conditionDataFactory
-     * @param \Magento\SalesRule\Api\Data\RuleLabelInterfaceFactory $ruleLabelFactory
-     * @param \Magento\Framework\Reflection\DataObjectProcessor $dataObjectProcessor
+     * @param RuleFactory $ruleFactory
+     * @param RuleInterfaceFactory $ruleDataFactory
+     * @param ConditionInterfaceFactory $conditionDataFactory
+     * @param RuleLabelInterfaceFactory $ruleLabelFactory
+     * @param DataObjectProcessor $dataObjectProcessor
      * @param Json $serializer Optional parameter for backward compatibility
      * @param RuleExtensionFactory|null $extensionFactory
      */
     public function __construct(
-        \Magento\SalesRule\Model\RuleFactory $ruleFactory,
-        \Magento\SalesRule\Api\Data\RuleInterfaceFactory $ruleDataFactory,
-        \Magento\SalesRule\Api\Data\ConditionInterfaceFactory $conditionDataFactory,
-        \Magento\SalesRule\Api\Data\RuleLabelInterfaceFactory $ruleLabelFactory,
-        \Magento\Framework\Reflection\DataObjectProcessor $dataObjectProcessor,
-        Json $serializer = null,
-        RuleExtensionFactory $extensionFactory = null
+        protected readonly RuleFactory $ruleFactory,
+        protected readonly RuleInterfaceFactory $ruleDataFactory,
+        protected readonly ConditionInterfaceFactory $conditionDataFactory,
+        protected readonly RuleLabelInterfaceFactory $ruleLabelFactory,
+        protected readonly DataObjectProcessor $dataObjectProcessor,
+        private ?Json $serializer = null,
+        private ?RuleExtensionFactory $extensionFactory = null
```

Comment:
The type of this argument has changed by mistake.
```suggestion
        private RuleExtensionFactory $extensionFactory = null,
```